examples/dev/triangle_edge.py

```diff
@@ -0,0 +1,259 @@
+import typing
+
+import numba
+import numpy as np
+
+import napari
+from napari.layers import Points, Shapes, Vectors
+from napari.layers.shapes._shapes_utils import generate_2D_edge_meshes
+
+
+def generate_regular_polygon(n, radius=1):
+    """
+    Generates the vertices of a regular n-sided polygon centered at the origin.
+
+    Parameters:
+    n (int): The number of sides (vertices).
+    radius (float): The radius of the circumscribed circle. Default is 1.
+
+    Returns:
+    np.ndarray: An array of shape (n, 2) containing the vertices coordinates.
+    """
+    angles = np.linspace(0, 2 * np.pi, n, endpoint=False)
+    return np.column_stack((radius * np.cos(angles), radius * np.sin(angles)))
+
+
+# @numba.njit
+def generate_order_vectors(path_, closed) -> np.ndarray:
+    """Generate the vectors tangent to the path."""
+    n = path_[1:] - path_[:-1]
+    norm = np.linalg.norm(n, axis=1, keepdims=True)
+    normals = n / norm
+    vec = np.empty((path_.shape[0], 2, 2))
+    vec[:, 0] = path_
+    vec[:-1, 1] = normals
+    if closed:
+        vec[-1, 1] = (path_[0] - path_[-1]) / np.linalg.norm(path_[-1] - path_[0])
+    else:
+        vec[-1, 1] = -vec[-2, 1]
+    return vec
+
+# @numba.njit
+def generate_miter_helper_vectors(normal_vector: np.ndarray) -> np.ndarray:
+    """Generate the miter helper vectors.
+
+    for each point on the path, the miter helper vectors are pairs of vectors
+    First vector is the normal vector scaled by 1/2
+    Second vector is the normal vector of previous point scaled by -1/2
+    """
+    vec2 = normal_vector.copy()
```

Comment:
Any particular reason that `vec2` and `vec3` are used instead of `vec1` and `vec2`?

Reply:
No. Only result of splitting functions of smaller one.